memory/memhotplug.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Abdul Haleem <abdhalee@linux.vnet.ibm.com>
+
+import os
+import glob
+import re
+import multiprocessing
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.utils.software_manager import SoftwareManager
+
+
+blocks_hotpluggable = []
+mem_path = '/sys/devices/system/memory'
+errorlog = ['WARNING: CPU:', 'Oops',
+            'Segfault', 'soft lockup',
+            'Unable to handle paging request',
+            'rcu_sched detected stalls',
+            'NMI backtrace for cpu',
+            'Call Trace:', 'WARNING: at',
+            'INFO: possible recursive locking detected',
+            'Kernel BUG at', 'Kernel panic - not syncing:',
+            'double fault:']
+
+
+def online(block):
+    try:
+        memory.hotplug(block)
+    except IOError:
+        print "memory%s : Resource is busy" % block
+        pass
+
+
+def offline(block):
+    try:
+        memory.hotunplug(block)
+    except IOError:
+        print "memory%s : Resource is busy" % block
+        pass
+
+
+def get_hotpluggable_blocks(path):
+    mem_blocks = []
+    blocks_total = glob.glob(path)
+    for mem_blk in blocks_total:
```

Comment:
can it be like 
`for mem_blk in glob.glob(path):` just want to eliminate extra variable

Reply:
done